contracts/Account.cairo

```diff
@@ -149,92 +167,183 @@ func is_valid_signature{
 end
 
 @external
-func execute{
+func __execute__{
         syscall_ptr : felt*, 
         pedersen_ptr : HashBuiltin*,
         range_check_ptr, 
         ecdsa_ptr: SignatureBuiltin*
     }(
-        to: felt,
-        selector: felt,
+        mcalls_len: felt,
+        mcalls: MCall*,
         calldata_len: felt,
         calldata: felt*,
         nonce: felt
     ) -> (response_len: felt, response: felt*):
     alloc_locals
 
     let (__fp__, _) = get_fp_and_pc()
-    let (_address) = get_contract_address()
+    let (tx_info) = get_tx_info()
     let (_current_nonce) = current_nonce.read()
 
     # validate nonce
     assert _current_nonce = nonce
 
-    local message: Message = Message(
-        _address,
-        to,
-        selector,
-        calldata,
-        calldata_size=calldata_len,
-        _current_nonce
-    )
+    # TMP: convert `MCall` to 'Call'
```

Comment:
Maybe it's worth adding a small comment on why it's temporal

Reply:
Done.